src/Components/Applications/Form/FormApplication.js

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useState } from 'react';
+import styles from './form.module.css';
+
+const FormApplication = () => {
+  const [position, setPositionName] = useState('');
+  const [company, setCompany] = useState('');
+  const [postulant, setPostulant] = useState('');
+  const [applicationState, setAppState] = useState("");
+
+   const onChangePosition = (event) => {
+    setPositionName(event.target.value);
+  };
+  const onChangeCompany = (event) => {
+    setCompany(event.target.value);
+  };
+  const onChangePostulant = (event) => {
+    setPostulant(event.target.value);
+  };
+  const onChangeAppState = (event) => {
+    setAppState(event.target.value);
+    console.log(event.target.value);
+  };
+
+  const params = new URLSearchParams(window.location.search);
+  const AppId = params.get('id');
+
+  useEffect(()=> {
+    if(AppId){
+      fetch(`${process.env.REACT_APP_API}/api/applications/id/${AppId}`)
+        .then((response) => {
+          if (response.status !== 200) {
+            return response.json().then(({ ErrMessage }) => {
+              throw new Error(ErrMessage);
+            });
+          }
+          return response.json();
+        })
+        .then((response) => {
+          setPositionName(response.positionId);
+          setCompany(response.companyId);
+          setPostulant(response.postulantId ? response.postulantId :"No id"); // Bad DB. Applications with no postulantID exist
+          setAppState(response.applicationState);
+        });
+    }
+  }, []);
+  console.log (position, company,postulant, applicationState );
+  const onSubmit = (event) => {
+    event.preventDefault();
+    let url;
+    const options = {
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        positionId: position,
+        companyId: company,
+        postulantId: postulant,
+        applicationState: applicationState
+      })
+    };
+
+    if (AppId === null) {
+      options.method = 'POST';
+      url = `${process.env.REACT_APP_API}/api/applications/add`;
+    } else {
+      options.method = 'PUT';
+      url = `${process.env.REACT_APP_API}/api/applications/update/${AppId}`;
+    }
+    console.log(url); 
+    console.log(options);
+    fetch(url, options).then((response) => {
+      if (response.status !== 200 && response.status !== 201){
+        return response.json().then(({ErrMessage}) => {
+          throw new Error(ErrMessage);
+        });
+      }
+      response.json();
```

Comment:
Is this needed here? I think you can just redirect on this line and remove the 2nd `.then`
```suggestion
      window.location.href = `/applications`;
```